toolbox/library/src/YarpRpc.cpp

```diff
@@ -0,0 +1,257 @@
+        {ParameterType::STRING, ParamIndex::RpcCommand, 1, 1, "RpcCommand"}
+    };
+
+    for (const auto& md : metadata) {
+        if (!blockInfo->addParameterMetadata(md)) {
+            bfError << "Failed to store parameter metadata";
+            return false;
+        }
+    }
+
+    return blockInfo->parseParameters(m_parameters);
+}
+
+bool YarpRpc::configureSizeAndPorts(BlockInformation* blockInfo)
+{
+    // INPUT
+    // =====
+    //
+    // 1) Trigger signal
+    //
+    // OUTPUT
+    // ======
+    //
+    // No outputs
+    //
+
+    const bool ok = blockInfo->setPortsInfo(
+        {
+            // Inputs
+            {InputIndex::TriggerSignal, Port::Dimensions{1}, Port::DataType::DOUBLE},
+        },
+        {
+            // Outputs
+        });
+
+    if (!ok) {
+        bfError << "Failed to configure input / output ports.";
+        return false;
+    }
+
+    return true;
+}
+
+bool YarpRpc::initialize(BlockInformation* blockInfo)
+{
+    if (!Block::initialize(blockInfo)) {
+        return false;
+    }
+
+    // PARAMETERS
+    // ==========
+
+    if (!YarpRpc::parseParameters(blockInfo)) {
+        bfError << "Failed to parse parameters.";
+        return false;
+    }
+
+    // Temp variable to store rpc command string
+    std::string command;
+
+    bool ok = true;
+    ok = ok && m_parameters.getParameter("RisingEdgeTrigger", pImpl->risingEdgeTrigger);
+    ok = ok && m_parameters.getParameter("PortName", pImpl->serverPortName);
+    ok = ok && m_parameters.getParameter("RpcCommand", command);
+
+    if (!ok) {
+        bfError << "Failed to read input parameters.";
+        return false;
+    }
+
+    // CLASS INITIALIZATION
+    // ====================
+
+    pImpl->network = std::make_unique<yarp::os::Network>();
+
+    if (!yarp::os::Network::initialized() || !yarp::os::Network::checkNetwork(5.0)) {
+        bfError << "YARP server wasn't found active.";
+        return false;
+    }
+
+    // Open a rpc client port
+    pImpl->clientPortName = "...";
+
+    if (!pImpl->rpcClientPort.open(pImpl->clientPortName)) {
+        bfError << "Error while opening yarp rpc server port.";
+        return false;
+    }
+
+    if (!yarp::os::Network::connect(pImpl->rpcClientPort.getName(), pImpl->serverPortName)) {
+        bfError << "Failed to connect " << pImpl->serverPortName << " to "
+                << pImpl->rpcClientPort.getName() << ".";
+        return false;
+    }
+
+    // Store rpc command in bottle
+    pImpl->rpcCommand.fromString(command);
+
+    return true;
+}
+
+bool YarpRpc::terminate(const BlockInformation* /*blockInfo*/)
+{
+    if(pImpl->rpcClientPort.isWriting()) {
+        bfWarning << "Interrupting rpc port";
+        pImpl->rpcClientPort.interrupt();
+    }
+
+    yarp::os::Network::disconnect(pImpl->rpcClientPort.getName(), pImpl->serverPortName);
+
+    // Close the port
+    pImpl->rpcClientPort.close();
+
+    return true;
```

Comment:
What would happen if at this stage the remote is no longer available? If we handle the timeout in the `output` method, we should be sure that the `terminate` returns true too. What I mean, just double check that this is the case, I am not sure what the `isWriting` method is supposed to return in this case for instance.